lib/integrations/base_driver.rb

```diff
@@ -0,0 +1,55 @@
+module Integrations
+  class BaseDriver
+    def initialize(integration)
+      @integration = integration
+    end
+
+    def self.available
+      true
+    end
+
+    def name
+      self.class.display_name
+    end
+
+    def type
+      self.class.type
+    end
+
+    def config
+      Integrations.config[type]
+    end
+
+    def applications
+      self.class.applications
+    end
+
+    def selected_application
+      self.class.selected_application
```

Comment:
Write some specs for these red lines here. The applications, selected_application and type methods.